src/submit_referendum.rs

```diff
@@ -138,104 +130,166 @@ pub(crate) async fn generate_calls(proposal_details: &ProposalDetails) -> Possib
+	//
+	// 1. Create a whitelist call on Asset Hub:
+	//
+	//    let whitelist_call =
+	//        KusamaAssetHubRuntimeCall::Whitelist(WhitelistCall::whitelist_call {
+	//            call_hash: H256(proposal_hash),
+	//    });
+	//
+	// 2. Create an XCM to send this from the Relay Chain:
+	//    let send_whitelist = KusamaRuntimeCall::XcmPallet(
+	//        PolkadotXcmCall::send {
+	//            dest: Location { parents: 0, interior: X1([Parachain(1000)]) },
+	//            message: vec![UnpaidExecution, Transact {call: whitelist_call, ..}],
+	//        }
+	//    );
+	//
+	// 3. Make a Fellowship referendum for `send_whitelist`.
+	//
+	// 4. Make a public referendum on Asset Hub.
 	let proposal_bytes = get_proposal_bytes(proposal_details.proposal.clone());
-	let proposal_call_info = CallInfo::from_bytes(&proposal_bytes, Network::Kusama);
+	let proposal_call_info = CallInfo::from_bytes(&proposal_bytes, Network::KusamaAssetHub);
 
 	let public_referendum_dispatch_time = match proposal_details.dispatch {
 		DispatchTimeWrapper::At(block) => DispatchTime::At(block),
 		DispatchTimeWrapper::After(block) => DispatchTime::After(block),
 	};
 
-	let whitelist_call =
-		CallInfo::from_runtime_call(NetworkRuntimeCall::Kusama(KusamaRuntimeCall::Whitelist(
-			WhitelistCall::whitelist_call { call_hash: H256(proposal_call_info.hash) },
-		)));
-	let preimage_for_whitelist_call = CallInfo::from_runtime_call(NetworkRuntimeCall::Kusama(
-		KusamaRuntimeCall::Preimage(PreimageCall::note_preimage { bytes: whitelist_call.encoded }),
+	let whitelist_call = CallInfo::from_runtime_call(NetworkRuntimeCall::KusamaAssetHub(
+		KusamaAssetHubRuntimeCall::Whitelist(WhitelistCall::whitelist_call {
+			call_hash: H256(proposal_call_info.hash),
+		}),
 	));
 
+	// This is what the Fellowship will actually vote on enacting.
+	let whitelist_over_xcm = CallInfo::from_runtime_call(NetworkRuntimeCall::Kusama(
+		KusamaRuntimeCall::XcmPallet(KusamaXcmCall::send {
+			dest: Box::new(VersionedLocation::V5(Location {
+				parents: 0,
+				interior: X1([Parachain(1000)]),
+			})),
+			message: Box::new(V5(Xcm(vec![
+				Instruction::UnpaidExecution {
+					weight_limit: WeightLimit::Unlimited,
+					check_origin: None,
+				},
+				Instruction::Transact {
+					origin_kind: OriginKind::Xcm,
+					fallback_max_weight: None,
+					call: DoubleEncoded { encoded: whitelist_call.encoded },
+				},
+			]))),
+		}),
+	));
+
+	// The actual Fellowship referendum submission.
 	let fellowship_proposal = CallInfo::from_runtime_call(NetworkRuntimeCall::Kusama(
-		KusamaRuntimeCall::FellowshipReferenda(ReferendaCall::submit {
+		KusamaRuntimeCall::FellowshipReferenda(KusamaReferendaCall::submit {
 			proposal_origin: Box::new(KusamaOriginCaller::Origins(KusamaOpenGovOrigin::Fellows)),
-			proposal: Lookup { hash: H256(whitelist_call.hash), len: whitelist_call.length },
-			enactment_moment: DispatchTime::After(10),
+			proposal: Inline(BoundedVec(whitelist_over_xcm.encoded)),
```

Comment:
Nice, better than submitting another preimage. But maybe a test or `ensure` to make sure it's under the `Inline` limit?

Reply:
Added an assertion with an error message in [a69d4dc](https://github.com/joepetrowski/opengov-cli/pull/53/commits/a69d4dc00718fba59d81efbeb03b011a107494b4)